lib/Google/RestApi/Auth/ServiceAccount.pm

```diff
@@ -0,0 +1,116 @@
+package Google::RestApi::Auth::ServiceAccount;
+
+use strict;
+use warnings;
+
+our $VERSION = '0.3';
+
+use 5.010_000;
+
+use autodie;
+use File::Basename;
+use Type::Params qw(compile_named);
+use Types::Standard qw(Str ArrayRef);
+use WWW::Google::Cloud::Auth::ServiceAccount;
+use YAML::Any qw(Dump);
+
+no autovivification;
+
+use Google::RestApi::Utils qw(config_file);
+
+use parent 'Google::RestApi::Auth';
+
+do 'Google/RestApi/logger_init.pl';
+
+sub new {
+  my $class = shift;
+
+  my $self = config_file(@_);
+  state $check = compile_named(
+    config_file        => Str, { optional => 1 },
+    parent_config_file => Str, { optional => 1 },  # only used internally
+    account_file       => Str,
+    scope              => ArrayRef[Str],
+  );
+  $self = $check->(%$self);
+
+  return bless $self, $class;
+}
+
+sub headers {
+  my $self = shift;
+  return $self->{headers} if $self->{headers};
+  my $access_token = $self->access_token();
+  $self->{headers} = [ Authorization => "Bearer $access_token" ];
+  return $self->{headers};
+}
+
+sub access_token {
+  my $self = shift;
+  return $self->{access_token} if $self->{access_token};
+
+  my $auth = WWW::Google::Cloud::Auth::ServiceAccount->new(
+    credentials_path => $self->account_file(),
+    # undocumented feature of WWW::Google::Cloud::Auth::ServiceAccount
+    scope            => join(' ', @{ $self->{scope} }),
+  );
```

Comment:
this $auth object should be stored and reused instead of the token it produced at one point in time.

Reply:
Sure, whatever makes sense for this.